<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>983</width>
    <height>623</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="openInstructionButton">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>550</y>
      <width>131</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Add Instructions</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clearMemoryButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>550</y>
      <width>291</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear Memory</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clearRegButton">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>550</y>
      <width>291</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear Register</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>150</y>
      <width>341</width>
      <height>391</height>
     </rect>
    </property>
    <property name="title">
     <string>Control</string>
    </property>
    <widget class="QPushButton" name="excuteButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>320</y>
       <width>321</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Execute</string>
     </property>
    </widget>
    <widget class="QPushButton" name="decodeButton">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>80</y>
       <width>71</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Decode</string>
     </property>
    </widget>
    <widget class="QPushButton" name="fetchButton">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>40</y>
       <width>71</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Fetch</string>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Program counter:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="pCounter">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>40</y>
       <width>51</width>
       <height>26</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="instructionDecode">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>80</y>
       <width>113</width>
       <height>26</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Instruction:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>71</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>OP Code:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="opCodeDisplay">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>130</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QTextEdit" name="encodedInsMessage">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>321</width>
       <height>141</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>130</y>
       <width>21</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>R</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="rDisplay">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>130</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="xDisplay">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>130</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>130</y>
       <width>21</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="yDisplay">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>130</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>130</y>
       <width>21</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QPushButton" name="runOneCycleButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>350</y>
       <width>161</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Run one Cycle</string>
     </property>
    </widget>
    <widget class="QPushButton" name="resetCounter">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>40</y>
       <width>51</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Reset</string>
     </property>
    </widget>
    <widget class="QPushButton" name="runUntilHaltButton">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>350</y>
       <width>151</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Run until halt</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_4">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>550</y>
      <width>201</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>How to use</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="memoryDisplay">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>291</width>
      <height>501</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>67</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Memory</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="registerDisplay">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>40</y>
      <width>291</width>
      <height>501</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>20</y>
      <width>67</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Register</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="screen">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>40</y>
      <width>341</width>
      <height>101</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>20</y>
      <width>67</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Screen</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>983</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
